source/common/stats/symbol_table_impl.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/stats/symbol_table.h"
+
+#include "common/common/assert.h"
+
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Stats {
+
+class SymbolTableImpl : public SymbolTable {
+  friend class StatNameImpl;
+
+public:
+  StatNamePtr encode(const std::string& name) override;
+
+  // For testing purposes only.
+  size_t size() const { return sizeof(SymbolTableImpl) + encode_map_.size() + decode_map_.size(); }
```

Comment:
size() seems like a useful and idiomatic API to include, but I would expect it to be the number of symbols.

It's not entirely clear to me what this particular impl is returning though; you should doc that.  the first sizeof returns a number of bytes, and the map size() calls return cardinalities which need to be multiplied by something to become bytes.